### PR TITLE
First implementation.

### PR DESCRIPTION
**Pull request by nathanschepers**
Only server setup works. Client management is WIP.

### CODE REVIEW

File: openvpn-setup.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    # clean the keys directory, generate certificates and keys
+    os.chdir(ca_output_dir)
+    key_dir = os.environ['KEY_DIR']
+    key_size = os.environ['KEY_SIZE']
+    dh_file = os.path.join(key_dir, 'dh' + key_size + '.pem')
+    hmac_file = os.path.join(key_dir, 'ta.key')
+
+    print('cleaning up before key generation (running ./clean-all)')
+    subprocess.call(['./clean-all'])
```
Comment (mgax) [inline openvpn-setup.py:107]: You can define a function to wrap subprocess details (e.g. https://github.com/liquidinvestigations/docker-arm/blob/b9a9377dc8b5853064485c7406152520cee2bbe8/upload#L7-L10):

```python
def run(args, **kwargs):
    args = [str(a) for a in args]
    print('+', ' '.join(args))
    return subprocess.run(args, check=True, **kwargs)
```

And call it like so:

```python
pkitool_env = {}
# ... populate pki_env with the output of `['bash', '-c', 'source ' + 'vars' + ' && env']`
run(['./clean-all'], env=pkitool_env)
```

Comment (mgax) [reply openvpn-setup.py:107]: And, instead of calling `os.chdir(...)`, pass a `cwd` argument to `run`, whcih gets passed to subprocess.

Comment (nathanschepers) [reply openvpn-setup.py:107]: when I turn all of the subprocess() calls into individual functions, do you want them to return the value of run?

Comment (mgax) [reply openvpn-setup.py:107]: Only if it's used for something.

Comment (nathanschepers) [reply openvpn-setup.py:107]: Client config looking good. Doesn't work with UDP but seems ok for TCP:

https://gist.github.com/alsegno/b1f14972ebc1d8c042a2bfb26d4bb6bf

More testing is needed, and revocation needs to be done.

**Comment by nathanschepers** [comment]
Ok some remaining questions:

1 - do you want all the extraneous comments in the config file templates or should i remove them so it's just the config options we need.

2 - how do we want to handle the linux-required options in the client config?

```
# NOTE: These should be uncommented for Linux clients
# script-security 2
# up /etc/openvpn/update-resolv-conf
# down /etc/openvpn/update-resolv-conf
```

**Pull request merged**